lib/broccoli/bundler.js

```diff
@@ -148,8 +149,9 @@ module.exports = class Bundler {
     options = options || {};
 
     let scriptOutputFiles = options.scriptOutputFiles;
+    let applicationAndDepsDebugTree = new BroccoliDebug(applicationAndDepsTree, 'bundler:application-and-dependencies');
```

Comment:
We should avoid this style of `BroccoliDebug` instantiation as it cannot be "zero cost".  Instead, we should do:

```js
constructor() {
  this._debugTree = BroccoliDebug.buildDebugCallback('bundler');
}

bundleJs(_applicationAndDepsTree, options) {
  let applicationAndDepsTree = this._debugTree(_applicationAndDepsTree, 'application-and-dependencies');
  // ...snip...
}
```

If written in this way, when no environment variable is set there are no additional trees created (the callback is simply `(tree) => tree`), however if the environment variable is set it will return the wrapped tree.